File: src.kotlin/alphaTab/src/commonMain/kotlin/alphaTab/core/ecmaScript/ArrayBuffer.kt
package alphaTab.core.ecmaScript

@ExperimentalUnsignedTypes
class ArrayBuffer {
    public val raw: UByteArray

    public constructor(size: Double) {
        raw = UByteArray(size.toInt())
    }

    public constructor(raw: UByteArray) {
        this.raw = raw
    }
}

File: packages/csharp/src/AlphaTab/Core/EcmaScript/ArrayBuffer.cs
namespace AlphaTab.Core.EcmaScript;

public class ArrayBuffer
{
    public byte[] Raw { get; }
    public double ByteLength => Raw.Length;

    public ArrayBuffer(double size)
    {
        Raw = new byte[(int) size];
    }
    public ArrayBuffer(byte[] raw)
    {
        Raw = raw;
    }
}
